apps/hash-ai-worker-ts/src/activities/flow-activities/research-entities-action/infer-facts-from-web-page-worker-agent.ts

```diff
@@ -483,38 +493,11 @@ export const inferFactsFromWebPageWorkerAgent = async (params: {
                 tool call unless there are no entities to infer from this page.
               `),
             };
-          } else if (
-            toolCall.name === "inferFactsFromWebPage" ||
-            toolCall.name === "inferFactsFromText"
-          ) {
-            const toolCallInput = toolCall.input as ToolCallArguments[
-              | "inferFactsFromWebPage"
-              | "inferFactsFromText"];
-
-            const accessedRemoteFile =
-              "fileUrl" in toolCallInput
-                ? state.filesQueried.find(
-                    ({ url: previouslyQueriedFileUrl }) =>
-                      previouslyQueriedFileUrl === toolCallInput.fileUrl,
-                  )
-                : undefined;
-
-            if ("fileUrl" in toolCallInput && !accessedRemoteFile) {
-              return {
-                ...toolCall,
-                output: dedent(`
-                  You did not previously query the PDF file at the provided fileUrl: ${toolCallInput.fileUrl}.
-                  You must first query the PDF file with the relevant query using the "queryPdf" tool,
-                    before inferring entities from its text content.
-                `),
-                isError: true,
-              };
-            }
+          } else if (toolCall.name === "inferFactsFromWebPage") {
+            const toolCallInput =
+              toolCall.input as ToolCallArguments["inferFactsFromWebPage"];
 
-            if (
-              "url" in toolCallInput &&
-              !haveSameTopLevelDomain(toolCallInput.url, input.url)
-            ) {
+            if (!haveSameTopLevelDomain(toolCallInput.url, input.url)) {
```

Comment:
Doesn't have to be done now but a reminder we were going to remove this restriction, and more reliably/consistently return a list of URLs visited to the coordinator agent

Reply:
Yes, have added a `@todo` and a note in the top-level ticket in https://github.com/hashintel/hash/pull/4593/commits/9285b5c33fe0981d11b8b5bad49cb431f5deaa2e